long double (a.k.a 128 bits floating point) data support

Hi there,

I tried to use "long double" typed variables with HighFive and got rejected by the compiler.

```
    H5Easy::File file("ggg.h5",H5Easy::File::Overwrite);
    long double lda = 1.17;
    H5Easy::dump(file, "/long_double_a", lda);
```

I've the following compiler error:
```
In file included from /usr/local/include/highfive/H5DataType.hpp:369,
                 from /usr/local/include/highfive/H5Attribute.hpp:15,
                 from /usr/local/include/highfive/bits/H5Annotate_traits.hpp:14,
                 from /usr/local/include/highfive/H5File.hpp:16,
                 from highFiveNoEasy.cpp:1:
/usr/local/include/highfive/bits/H5DataType_misc.hpp: In instantiation of ‘HighFive::AtomicType<T>::AtomicType() [with T = long double]’:
/usr/local/include/highfive/bits/H5DataType_misc.hpp:428:12:   required from ‘HighFive::DataType HighFive::create_datatype() [with T = long double]’
/usr/local/include/highfive/bits/H5DataType_misc.hpp:436:36:   required from ‘HighFive::DataType HighFive::create_and_check_datatype() [with T = long double]’
/usr/local/include/highfive/bits/H5Node_traits_misc.hpp:63:57:   required from ‘HighFive::DataSet HighFive::NodeTraits<Derivate>::createDataSet(const string&, const HighFive::DataSpace&, const DataSetCreateProps&, const DataSetAccessProps&, bool) [with Type = long double; Derivate = HighFive::File; std::string = std::__cxx11::basic_string<char>; HighFive::DataSetCreateProps = HighFive::PropertyList<HighFive::PropertyType::DATASET_CREATE>; HighFive::DataSetAccessProps = HighFive::PropertyList<HighFive::PropertyType::DATASET_ACCESS>]’
highFiveNoEasy.cpp:25:93:   required from here
/usr/local/include/highfive/bits/H5DataType_misc.hpp:184:57: error: static assertion failed: Type not supported
  184 |     static_assert(details::inspector<T>::recursive_ndim > 0, "Type not supported");
      |                                          ~~~~~~~~~~~~~~~^~~
```

As far as I understand HighFive does not support "long double" data types. Am I right?

If this is the case, is there a plan to add such a support to HighFives?

Thanks,
Ron

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -31,7 +31,7 @@ jobs:
     runs-on: ${{matrix.os}}
     strategy:
       matrix:
-        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        os: [ubuntu-18.04, ubuntu-20.04]
 
     steps:
     - uses: actions/checkout@v2

--- a/include/highfive/bits/H5DataType_misc.hpp
+++ b/include/highfive/bits/H5DataType_misc.hpp
@@ -126,7 +126,7 @@ inline AtomicType<unsigned long long>::AtomicType() {
     _hid = H5Tcopy(H5T_NATIVE_ULLONG);
 }
 
-// float and double mapping
+// float, double and long double mapping
 template <>
 inline AtomicType<float>::AtomicType() {
     _hid = H5Tcopy(H5T_NATIVE_FLOAT);
@@ -135,6 +135,11 @@ inline AtomicType<float>::AtomicType() {
 template <>
 inline AtomicType<double>::AtomicType() {
     _hid = H5Tcopy(H5T_NATIVE_DOUBLE);
+}
+
+template <>
+inline AtomicType<long double>::AtomicType() {
+    _hid = H5Tcopy(H5T_NATIVE_LDOUBLE);
 }
 
 // boolean mapping

--- a/tests/unit/tests_high_five_base.cpp
+++ b/tests/unit/tests_high_five_base.cpp
@@ -247,8 +247,8 @@ BOOST_AUTO_TEST_CASE(HighFiveGroupAndDataSet) {
 BOOST_AUTO_TEST_CASE(HighFiveExtensibleDataSet) {
     const std::string FILE_NAME("create_extensible_dataset_example.h5");
     const std::string DATASET_NAME("dset");
-    constexpr double t1[3][1] = {{2.0}, {2.0}, {4.0}};
-    constexpr double t2[1][3] = {{4.0, 8.0, 6.0}};
+    constexpr long double t1[3][1] = {{2.0}, {2.0}, {4.0}};
+    constexpr long double t2[1][3] = {{4.0, 8.0, 6.0}};
 
     {
         // Create a new file using the default property lists.
@@ -263,7 +263,7 @@ BOOST_AUTO_TEST_CASE(HighFiveExtensibleDataSet) {
 
         // Create the dataset
         DataSet dataset = file.createDataSet(DATASET_NAME, dataspace,
-                                             AtomicType<double>(), props);
+                                             AtomicType<long double>(), props);
 
         // Write into the initial part of the dataset
         dataset.select({0, 0}, {3, 1}).write(t1);
@@ -290,7 +290,7 @@ BOOST_AUTO_TEST_CASE(HighFiveExtensibleDataSet) {
 
         DataSet dataset_absolute = file.getDataSet("/" + DATASET_NAME);
         const auto dims = dataset_absolute.getSpace().getDimensions();
-        double values[4][6];
+        long double values[4][6];
         dataset_absolute.read(values);
         BOOST_CHECK_EQUAL(4, dims[0]);
         BOOST_CHECK_EQUAL(6, dims[1]);

--- a/tests/unit/tests_high_five_base.cpp
+++ b/tests/unit/tests_high_five_base.cpp
@@ -247,8 +247,8 @@ BOOST_AUTO_TEST_CASE(HighFiveGroupAndDataSet) {
 BOOST_AUTO_TEST_CASE(HighFiveExtensibleDataSet) {
     const std::string FILE_NAME("create_extensible_dataset_example.h5");
     const std::string DATASET_NAME("dset");
-    constexpr long double t1[3][1] = {{2.0}, {2.0}, {4.0}};
-    constexpr long double t2[1][3] = {{4.0, 8.0, 6.0}};
+    constexpr long double t1[3][1] = {{2.0l}, {2.0l}, {4.0l}};
+    constexpr long double t2[1][3] = {{4.0l, 8.0l, 6.0l}};
 
     {
         // Create a new file using the default property lists.

